src/main/java/de/zalando/aruha/nakadi/controller/EventPublishingController.java

```diff
@@ -75,17 +85,16 @@ private ResponseEntity response(final EventPublishResult result) {
         }
     }
 
-    private ResponseEntity doWithMetrics(final String eventTypeName, final long startingNanos,
-                                         final EventProcessingTask task) throws NakadiException {
+    private ResponseEntity doWithTimerMetric(final EventTypeMetrics eventTypeMetrics, final long startingNanos,
```

Comment:
why pass a start time and calculate the delta yourselve if, doWithTimerMetric, receives the metric and from it could extract a timer (c  = timer.time() then c.stop()) ?